Source/ACE/Entity/Player.cs

```diff
@@ -765,15 +770,96 @@ public void SaveOptions()
+        /// </summary>
+        public void SetLifestoneUseCharacterPosition()
+        {
+            // Saves the current player position after converting from a Position Object, to a CharacterPosition object
+            character.SetCharacterPositions(CharacterPositionType.LifestoneUsed, CharacterPositionExtensions.positionToCharacterPosition(character.Id, Session.Player.Position, CharacterPositionType.LifestoneUsed));
+        }
+
+        /// <summary>
+        /// Set's the tied lifestone for use casting the Lifestone Recall spell
+        /// </summary>
+        public void SetLifestoneTiedCharacterPosition(CharacterPosition newPosition)
+        {
+            // TODO: Set too destination of the tied portal
+            character.SetCharacterPositions(CharacterPositionType.LifestoneUsed, newPosition);
+        }
+
+        /// <summary>
+        /// Set's the last portal used
+        /// </summary>
+        public void SetPortalRecallCharacterPosition(CharacterPosition newPosition)
+        {
+            // TODO : Save the Portal Recall position information from the destination of a portal
+            character.SetCharacterPositions(CharacterPositionType.PortalRecall, newPosition);
+        }
+
+        /// <summary>
+        /// Set's the Primary tied portal CharacterPosition for use with the portal recall spells
+        /// </summary>
+        public void SetPrimaryPortalRecallCharacterPosition(CharacterPosition newPosition)
+        {
+            // TODO : Save the Portal Recall position information from the destination of a portal
+            character.SetCharacterPositions(CharacterPositionType.PrimaryPortalRecall, newPosition);
+        }
+
+        /// <summary>
+        /// Set's the Secondary tied portal CharacterPosition for use with the portal recall spells
+        /// </summary>
+        public void SetSecondaryPortalRecallCharacterPosition(CharacterPosition newPosition)
+        {
+            // TODO : Save the Portal Recall position information from the destination of a portal
+            character.SetCharacterPositions(CharacterPositionType.SecondaryPortalRecall, newPosition);
+        }
+
+        /// <summary>
+        /// Set's allegiance recall position
+        /// </summary>
+        public void SetAllegianceCharacterPosition(CharacterPosition newPosition)
+        {
+            // TODO : Save the Allegiance Recall position information from the destination of a bind stone
+            character.SetCharacterPositions(CharacterPositionType.AllegianceHometown, newPosition);
+        }
+
+        /// <summary>
+        /// Set's mansion recall position
+        /// </summary>
+        public void SetMansionCharacterPosition(CharacterPosition newPosition)
+        {
+            // TODO : Save the Mansion Recall position information from the destination of a house box
+            character.SetCharacterPositions(CharacterPositionType.MansionRecall, newPosition);
+        }
+

```

Comment:
Not sure we'd set Allegiance Hometown and Mansion Recall positions to each character.. Mansion recall is essentially a copy of monarch's house recall if house is a villa or mansion, otherwise it didn't work or errored, not sure what it did here. I assume it would be a check if character is in allegiance, followed by confirming house type, then execution or returning error.

Hometown recall might be stored in the allegiance information somewhere, i haven't seen it yet however.